app/unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -125,7 +129,7 @@ public PrototypeFactory getPrototypeFactory(Context c) {
      */
     private static void initFactoryClassList() {
         if (factoryClassNames.isEmpty()) {
-            String baseODK = BuildConfig.BUILD_DIR + "/intermediates/classes/commcare/debug/";
+            String baseODK = BuildConfig.BUILD_DIR + "/intermediates/javac/commcareDebug/compileCommcareDebugJavaWithJavac/classes/";
```

Comment:
Is it possible to investigate whether there are better ways for us to estimate where these files come from? I'm totally fine if the answer is 'no' or 'it's a lot of work to even check,' but it worries me to keep relying on this hard coded of a setup

Reply:
I looked for it but could not find anything straightforward. Some people seems to define multiple paths in such cases hoping only one of them will exist at a time. Though I don't think it changes for a given version of AGP and therefore don't see value in relying on multiple paths either.

Reply:
Classic Java / Gradle - "Hyper generic until you need to hardcode 50 paths" 

this is awesome work, though, thanks for going through the deep slog of getting things compatible with new versions! 

Let me know when you think it's good-to-go by the way, and I'll spin up a test on my machine. Trying it out on a windows environment has illustrated some overfitting previously

Reply:
Hey yeah , It would be great if you can test it on your end. I have been able to run successful clean builds on Jenkins and on my own machine. So it's very much good to go from my side. I think you will need to update your AS thoughl. I am using AS 3.4 on my end though seems like AS 3.5 was just released.

Reply:
Sounds like as good a time to jump in as any. I'll update AS today and try this branch